hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconScmMetadataManagerImpl.java

```diff
@@ -0,0 +1,219 @@
+        DBStoreBuilder.newBuilder(ozoneConfiguration)
+            .setName(dbFile.getName())
+            .setPath(dbFile.toPath().getParent());
+    for (DBColumnFamilyDefinition columnFamily :
+        definition.getColumnFamilies()) {
+      dbStoreBuilder.addTable(columnFamily.getName());
+      dbStoreBuilder.addCodec(columnFamily.getKeyType(),
+          columnFamily.getKeyCodec());
+      dbStoreBuilder.addCodec(columnFamily.getValueType(),
+          columnFamily.getValueCodec());
+    }
+    return dbStoreBuilder.build();
+  }
+
+  /**
+   * Replace existing DB instance with new one.
+   *
+   * @param dbFile new DB file location.
+   */
+  private void initializeRdbStoreWithFile(File dbFile)
+      throws IOException {
+    try {
+      DBStore newStore = createDBAndAddSCMTablesAndCodecs(
+          dbFile, new ReconSCMSnapshotDBDefinition());
+
+      sequenceIdGen.reinitialize(
+          ReconSCMSnapshotDBDefinition.SEQUENCE_ID.getTable(newStore));
+      ozoneStorageContainerManager.getPipelineManager().reinitialize(
+          ReconSCMSnapshotDBDefinition.PIPELINES.getTable(newStore));
+      ozoneStorageContainerManager.getContainerManager().reinitialize(
+          ReconSCMSnapshotDBDefinition.CONTAINERS.getTable(newStore));
+
+      setStore(newStore);
+      ozoneStorageContainerManager.setStore(newStore);
+      LOG.info("Created SCM DB handle from snapshot at {} and sequence Id - {}.", dbFile.getAbsolutePath(),
+          getLastSequenceNumberFromDB());
+    } catch (IOException ioEx) {
+      LOG.error("Unable to initialize Recon SCM DB snapshot store.", ioEx);
+    }
+    if (getStore() != null) {
+      initializeScmTables();
+    }
+  }
+
+  /**
+   * Refresh the DB instance to point to a new location. Get rid of the old
+   * DB instance.
+   *
+   * @param newDbLocation New location of the SCM Snapshot DB.
+   */
+  @Override
+  public void updateScmDB(File newDbLocation) throws IOException {
+    DBStore current = ozoneStorageContainerManager.getStore();
+    if (null != current) {
+      File oldDBLocation = current.getDbLocation();
+      if (oldDBLocation.exists()) {
+        LOG.info("Cleaning up old SCM snapshot db at {}.",
+            oldDBLocation.getAbsolutePath());
+        FileUtils.deleteDirectory(oldDBLocation);
+      }
+    }
+    try {
+      initializeRdbStoreWithFile(newDbLocation);
+    } finally {
+      // Always close DBStore if it's replaced.
+      if (current != null && current != ozoneStorageContainerManager.getStore()) {
+        current.close();
+      }
+    }
+  }
+
+  /**
+   * Get SCM metadata RocksDB's latest sequence number.
+   * @return latest sequence number.
+   */
+  @Override
+  public long getLastSequenceNumberFromDB() {
+    RDBStore rocksDBStore = (RDBStore) getStore();
+    if (null == rocksDBStore) {
+      return 0;
+    } else {
+      try {
+        LOG.error("rocksDBStore.getDbLocation().getAbsolutePath(): {}", rocksDBStore.getDbLocation().getAbsolutePath());
```

Comment:
We can revise the message to clearly explain what the error is, why it occurred, and possibly how to resolve it. For example:
```suggestion
LOG.error("Failed to access RocksDB at location: {}. Ensure the database path is correct and accessible.", rocksDBStore.getDbLocation().getAbsolutePath());
```

Reply:
Oh, I think this log was left to be removed. It was added for some debugging purpose, forgot to remove. Will remove this.

Reply:
Should we consider including logging message in case the user cannot access the db sue to permission issues?

Reply:
> Should we consider including logging message in case the user cannot access the db due to permission issues?

Yes.

> Should we consider including logging message in case the user cannot access the db due to permission issues?

Yes, added the error log.